src/common/error.rs

```diff
@@ -1,12 +1,22 @@
 use crate::pam::PamError;
-use std::{fmt, path::PathBuf};
+use std::{borrow::Cow, fmt, path::PathBuf};
 
 #[derive(Debug)]
 pub enum Error {
+    Silent,
```

Comment:
Just curious, why do we need this?

Reply:
`sudo --list some-command` triggers this error path when the invoking user cannot run `sudo some-command` due to sudoers policy. ogsudo fails silently with exit code 1 in that case and we are doing the same right now but we could improve diagnostics in the future and e.g. report back "user ferris cannot run `sudo some-command`( as other-user) on some-hostname`. That improvement would involve repurposing this error variant.

tl;dr it has been added to allow improving diagnostics in the future (if we decide to do so)